common/flatpak-run.c

```diff
@@ -1622,13 +1622,30 @@ job_removed_cb (SystemdManager *manager,
     g_main_loop_quit (data->main_loop);
 }
 
+static GString *
+systemd_unit_name_escape (const gchar *in)
+{
+  /* Adapted from systemd source */
+  GString * const ret = g_string_sized_new (strlen (in) * 4);
```

Comment:
Why make it size * 4? Thats almost always going to allocate too much. The point of GString is that it reallocs nicely.

Reply:
that is indeed the worst case.